src/main/scala/edu/ie3/simona/agent/grid/GridAgentMessages.scala

```diff
@@ -0,0 +1,189 @@
+ */
+
+package edu.ie3.simona.agent.grid
+
+import edu.ie3.simona.agent.grid.GridAgentData.GridAgentInitData
+import edu.ie3.simona.agent.grid.GridAgentMessages.SlackVoltageResponse.ExchangeVoltage
+import edu.ie3.simona.ontology.messages.PowerMessage.PowerResponseMessage
+import edu.ie3.simona.ontology.messages.{Activation, PowerMessage}
+import edu.ie3.simona.scheduler.ScheduleLock.ScheduleKey
+import org.apache.pekko.actor.typed.ActorRef
+import squants.electro.ElectricPotential
+
+import java.util.UUID
+
+/** Defines all messages that can be received by a [[GridAgent]] without the
+  * need for an adapter.
+  */
+object GridAgentMessages {
+
+  /** GridAgent initialization data can only be constructed once all GridAgent
+    * actors are created. Thus, we need an extra initialization message.
+    *
+    * @param gridAgentInitData
+    *   The initialization data
+    */
+  final case class CreateGridAgent(
+      gridAgentInitData: GridAgentInitData,
+      unlockKey: ScheduleKey,
+  ) extends GridAgent.InternalRequest
+
+  /** Trigger used inside of [[edu.ie3.simona.agent.grid.DBFSAlgorithm]] to
+    * execute a power flow calculation
+    *
+    * @param tick
+    *   current tick
+    */
+  final case class DoPowerFlowTrigger(tick: Long, currentSweepNo: Int)
+      extends GridAgent.InternalRequest
+
+  /** Trigger used inside of [[edu.ie3.simona.agent.grid.DBFSAlgorithm]] to
+    * activate the superior grid agent to check for deviation after two sweeps
+    * and see if the power flow converges
+    *
+    * @param tick
+    *   current tick
+    */
+  final case class CheckPowerDifferencesTrigger(tick: Long)
+      extends GridAgent.InternalRequest
+
+  /** Trigger used inside of [[edu.ie3.simona.agent.grid.DBFSAlgorithm]] to
+    * trigger the [[edu.ie3.simona.agent.grid.GridAgent]] s to prepare
+    * themselves for a new sweep
+    *
+    * @param tick
+    *   current tick
+    */
+  final case class PrepareNextSweepTrigger(tick: Long)
+      extends GridAgent.InternalRequest
+
+  /** Trigger used inside of [[edu.ie3.simona.agent.grid.DBFSAlgorithm]] to
+    * indicate that a result has been found and each
+    * [[edu.ie3.simona.agent.grid.GridAgent]] should do it's cleanup work
+    *
+    * @param tick
+    *   current tick
+    */
+  final case class FinishGridSimulationTrigger(tick: Long)
+      extends GridAgent.InternalRequest
+
+  /** Wrapper for activation values
+    *
+    * @param activation
+    *   the tick
+    */
+  final case class WrappedActivation(activation: Activation)
+      extends GridAgent.InternalRequest
+
+  /** Wrapper for any [[GridAgent.Response]] from [[GridAgent]]s.
+    * @param response
+    *   received response
+    */
+  final case class WrappedResponse(response: GridAgent.Response)
+      extends GridAgent.InternalRequest
+
+  /** Wrapper for [[PowerMessage]]s.
+    * @param msg
+    *   the received power message
+    */
+  final case class WrappedPowerMessage(msg: PowerMessage)
```

Comment:
There should be no need to wrap power messages. They should be either always received by GridAgents or always received by ParticipantAgents.

For example, `RequestAssetPowerMessage` is always received by a ParticipantAgent. It should thus be part of the messages received by ParticipantAgents. 

`AssetPowerChangedMessage` is always received by GridAgents and should thus extend `GridAgent.Request` (at least transitively).